codility_training/lessons.lesson10.PrimeAndCompositeNumbers.MinPerimeterRectangle/NowonLee.js

```diff
@@ -0,0 +1,12 @@
+function solution(N) {
+    let l = []
+    for(let i = 1;i*i<=N;i++){
+        if(N%i == 0) l.push(i)
+    }
+    const nums = l.map(el => {
+        return 2*(el + (N/el))
+    })
+    return Math.min(...nums)
```

Comment:
```suggestion
    return Math.min(...l.map(el => 2 * (el + (N / el ))))
```
이렇게 할 수도 있고, 더 좋은 방법은 reduce를 사용하는 것입니다.

```suggestion
    return l.reduce((min, el) => Math.min( min, 2 * ( el + ( N / el ) ) ), 0)
```

이러면 한 번의 반복으로 끝내기 가능!